cibuildwheel/__main__.py

```diff
@@ -183,15 +199,11 @@ def main() -> None:
     # This needs to be passed on to the docker container in linux.py
     os.environ['CIBUILDWHEEL'] = '1'
 
-    if not any((package_dir / name).exists()
```

Comment:
That's funny. Is there a reason we only checked the existence of one of these files so late (after already trying to parse a bunch of options) ?

If so, we should keep this order. If not, well, good to move it up, maybe. But worth thinking what's the logical precedence of failure to the user. I.e. if both `CIBW_ENVIRONMENT` is malformed and there's no `setup.py`, what error is the most severe one, that you expect to be reported? I think I agree with `python_package` not being a Python package is more important to first report :-)